### ISSUE TITLE
Add support for a Wix.toml manifest

### ISSUE THREAD

**Issue by volks73** [volks73/cargo-wix#20]
If a `Wix.toml` manifest exists in the root of the project or in the `wix` folder, then the fields within this file are used when building, compiling, linking, and signing. When the `--init` flag is used, a `Wix.toml` file is created in the `wix` folder as part of the initialization of the wix sub-project.

**Comment by volks73** [comment]
Instead of a Wix.toml file in the `wix` folder, it might be better to just add a section to the Cargo.toml for the project. The section can be named something like `[wix]`. This appears to be what the [cargo-deb](https://github.com/mmstick/cargo-deb) project does. I would like to be consistent as possible with other cargo subcommands.

**Comment by volks73** [comment]
This is most likely a duplicate of #20. The sub-command currently (v0.0.5) does not support a sub-command specific manifest file or Cargo.toml section, but I like all of these ideas and suggests. Please continue this discussion in #20.

Additionally,

>Also, it would be great, if you could support a "nice name". Cargo will complain if you capitalize the package name (which is OK), but for a user it's often more friendly to have a capitalized name

This is currently possible with the `-p,--product-name` command line option. Use the `cargo wix --help` command for more information and the [documentation](https://volks73.github.io/cargo-wix/cargo_wix/index.html#concepts) also provides some additional information about changing the product name:

>The cargo wix subcommand use the package name for the product name. The default install location is at C:\Program Files\[Product Name], where [Product Name] is replaced with the product name determined from the package name. This can be overridden with the -p,--product-name option. The binary name, which is the name field for the [[bin]] section, is used for the executable file name, i.e. "name.exe". This can also be overridden using the -b,--bin-name option. The package description is used in multiple places for the installer, including the text that appears in the blue UAC dialog when using a signed installer. This can be overridden using the -d,--description option.

If and when a manifest-based configuration system is implemented, a `product-name` field will be absolutely be included.

**Comment by fschutt** [comment]
If you could somehow support localization files (in the `Wix.toml` or a `package.metadata.wix` section), that'd be great. I.e. I have a `main_en-us.wxl` file, but I currently don't see any way of passing that localization information to light.

**Comment by volks73** [comment]
>If you could somehow support localization files...

I have also created #52 because it will be faster and easier to add support through the CLI instead of through the manifest at the moment.

**Comment by volks73** [comment]
The [cargo-deb](https://github.com/mmstick/cargo-deb) uses the Cargo.toml file and a `[package.metadata.deb]` section, so I am leaning towards the `[package.metadata.wix]` section in Cargo.toml implementation as opposed to a `Wix.toml` file.

**Comment by fschutt** [comment]
Yeah, that'd be nice.

However, forget about my "put everything in the Cargo.toml file" thing. I've discovered that `.wxs` files are much more complex that I initially thought. So in the end, you have to check in the `.wxs` file into version control no matter what - unless you want to port each and every feature from `.wxs` to `.toml`, which would be unnecessary.

**Comment by volks73** [comment]
>...you have to check in the .wxs file into version control no matter what...

I cannot find the article or documentation, but I remember reading somewhere that it is best practice to keep the `.wxs` file under version control and to think about it as being part of the source code. 

>...unless you want to port each and every feature from .wxs to .toml, which would be unnecessary.

Uh, no. You are correct. I do not want to port all of WiX into this sub-command. I would like this sub-command to easily generate a `.wxs` file that could be used as a bare minimum for creating an installer with the WiX Toolset, or be used as a template for adding more complex customizations and/or features to a Rust project's installer. 

This is a little convoluted at the moment given that the sub-command supports changing the product name, description, etc. at the command line using the XML preprocessor variables. Maybe these options and reading the Cargo.tom values should only be used with the `--init` flag? The idea being, a user would generate a base `.wxs` file generated from the contents of the Cargo.toml file with the `--init` flag and anything "tailoring" options, like changing the product name, and then the remaining functionality of this sub-command just focuses passing the necessary CLI options and flags to the compiler (candle.exe) and linker (light.exe). The `[package.metadata.wix]` section in the Cargo.toml file would have fields that match the flags and options used for all non-init/template building commands.

**Comment by volks73** [comment]
Implemented as of 6a2fdd2598292666f861cc75db5788260bade444

All of the default _create_ sub-command CLI options can be specified in a `[package.metadata.wix]` section of the Cargo.toml file. For example:
```toml
[package.metadata.wix]
culture = "FrFr"
input = "Path/to/WIX/Source/File.wxs"
locale = "Path/to/WIX/Localization/File.wxl"
name = "example"
no-build = false
output = "Path/and/file/name/for/installer.msi"
version = "2.1.0"
```
Note, the CLI options will override any values in the metadata section.

This is only implemented for the default _create_ sub-command. A number of other features are dependent on this feature, so I thought it was time to finally implement it. I would still like to do some more tests before I create a new release (v0.2.0), but the feature is at least available.
